src/source/tile.js

```diff
@@ -769,7 +804,17 @@ class Tile {
                 const x = i * step;
                 vertices.emplaceBack(x, y);
 
-                const ecef = tileCoordToECEF(x, y, id);
+                let ecef = tileCoordToECEF(x, y, id);
+
+                // When in globe-to-Mercator transition, interpolate between globe and Mercator positions in ECEF
+                if (worldToECEFMatrix) {
+                    const tileCount = 1 << id.z;
+                    const mercatorX = (x / EXTENT + id.x) / tileCount + wrap;
+                    const mercatorY = (y / EXTENT + id.y) / tileCount;
+                    const mercatorPos = [mercatorX * tr.worldSize, mercatorY * tr.worldSize, 0];
+                    vec3.transformMat4(mercatorPos, mercatorPos, worldToECEFMatrix);
+                    ecef = interpolateArray(ecef, mercatorPos, phase);
+                }
```

Comment:
Maybe share that with this [line](https://github.com/mapbox/mapbox-gl-js/pull/12118/files#diff-3dd84d28ccd48c2fc3addeef384f5ae733241d8236a6fcaa1e5fd7b0f2ce6835R753)?